pydicom/pixel_data_handlers/rle_handler.py

```diff
@@ -0,0 +1,171 @@
+                                  "Cannot return array. Pydicom might "
+                                  "be able to convert the pixel data "
+                                  "using GDCM if it is installed.")
+    if not have_numpy:
+        msg = ("The Numpy package is required to use pixel_array, and "
+               "numpy could not be imported.")
+        raise ImportError(msg)
+
+    if 'PixelData' not in dicom_dataset:
+        raise TypeError("No pixel data found in this dataset.")
+
+    # Make NumPy format code, e.g. "uint16", "int32" etc
+    # from two pieces of info:
+    # dicom_dataset.PixelRepresentation -- 0 for unsigned, 1 for signed;
+    # dicom_dataset.BitsAllocated -- 8, 16, or 32
+    if dicom_dataset.PixelRepresentation == 0:
+        format_str = 'uint{}'.format(dicom_dataset.BitsAllocated)
+    elif dicom_dataset.PixelRepresentation == 1:
+        format_str = 'int{}'.format(dicom_dataset.BitsAllocated)
+    else:
+        format_str = 'bad_pixel_representation'
+    try:
+        numpy_format = numpy.dtype(format_str)
+    except TypeError:
+        msg = ("Data type not understood by NumPy: "
+               "format='{}', PixelRepresentation={}, "
+               "BitsAllocated={}".format(
+                   format_str,
+                   dicom_dataset.PixelRepresentation,
+                   dicom_dataset.BitsAllocated))
+        raise TypeError(msg)
+
+    print("Decoding RLE")
+    UncompressedPixelData = bytearray()
+
+    if ('NumberOfFrames' in dicom_dataset and
+            dicom_dataset.NumberOfFrames > 1):
+
+        CompressedPixelDataSeq = pydicom.encaps.decode_data_sequence(
+            dicom_dataset.PixelData)
+
+        for frame in CompressedPixelDataSeq:
+            decompressed_frame = rle_decode_frame(frame,
+                                                  Rows=dicom_dataset.Rows,
+                                                  Columns=dicom_dataset.Columns,
+                                                  SamplesPerPixel=dicom_dataset.SamplesPerPixel,
+                                                  BitsAllocated=dicom_dataset.BitsAllocated)
+
+            UncompressedPixelData.extend(decompressed_frame)
+
+    else:
+
+        CompressedPixelData = pydicom.encaps.defragment_data(
+            dicom_dataset.PixelData)
+
+        decompressed_frame = rle_decode_frame(CompressedPixelData,
+                                              Rows=dicom_dataset.Rows,
+                                              Columns=dicom_dataset.Columns,
+                                              SamplesPerPixel=dicom_dataset.SamplesPerPixel,
+                                              BitsAllocated=dicom_dataset.BitsAllocated)
+
+        UncompressedPixelData.extend(decompressed_frame)
+
+    pixel_array = numpy.frombuffer(UncompressedPixelData, numpy_format)
+
+    return pixel_array
+
+
+def rle_decode_frame(d, Rows, Columns, SamplesPerPixel, BitsAllocated):
+    rle_start = 0
```

Comment:
Why did you define `rle_start` and `rle_len` when they never change and pretty much just take up character space? rle_start is always 0 so it can basically just be removed from all the places you use it.